source/MaterialXCore/Element.h

```diff
@@ -1150,6 +1150,22 @@ class ValueElement : public TypedElement
         return getAttribute(UNITTYPE_ATTRIBUTE);
     }
 
+    /// @}
+    /// @name Uniform attribute
+    /// @{
+
+    /// Set the uniform attribute flag on this element.
+    void setIsUniform(bool value)
+    {
+        setTypedAttribute<bool>(UNIFORM_ATTRIBUTE, value);
+    }
+
+    /// The the uniform attribute flag for this element.
+    bool getIsUniform() const
```

Comment:
This superscedes any type RTTI check for isA<Parameter> to see if something is a uniform or not.